main.go

```diff
@@ -269,12 +329,65 @@ func _main() error {
 		return err
 	}
 	defer os.RemoveAll(tmpDir)
+	if cache {
+		tmpDir = "dsq-cache"
+	}
+
+	settings := *runner.DefaultSettings
+	ec := runner.NewEvalContext(settings, tmpDir)
+
+	if cache {
+		// sqlite file is present on disk
+		if info, err := os.Stat(getCachedDBPath(project.Id)); err != nil || info.Size() == 0 {
+			file, err := openTruncate(getCachedDBPath(project.Id))
+			if err != nil {
+				return err
+			}
+			file.Close()
+		} else {
+			settings.CacheMode = true
+			ec = runner.NewEvalContext(settings, tmpDir)
+		}
+
+		notPresentFiles := []string{}
+		for i := 0; i < len(files); i++ {
```

Comment:
I am pretty sure there's a way to do this without duplicating all of this code. Something like just adding the file name read from stdin to the list of `files`.